domain/learningandworkprogress/src/main/kotlin/uk/gov/justice/digital/hmpps/domain/learningandworkprogress/review/service/ReviewService.kt

```diff
@@ -31,4 +34,71 @@ class ReviewService(
+      ReviewScheduleCalculationRule.PRISONER_READMISSION
+    } else if (isTransfer) {
+      ReviewScheduleCalculationRule.PRISONER_TRANSFER
+    } else {
+      when (sentenceType) {
+        SentenceType.REMAND -> ReviewScheduleCalculationRule.PRISONER_ON_REMAND
+        SentenceType.CONVICTED_UNSENTENCED -> ReviewScheduleCalculationRule.PRISONER_UN_SENTENCED
+        SentenceType.INDETERMINATE_SENTENCE -> ReviewScheduleCalculationRule.INDETERMINATE_SENTENCE
+
+        else -> reviewScheduleCalculationRuleBasedOnTimeLeftToServe(releaseDate!!)
+      }
+    }
+
+  fun calculateReviewWindow(reviewScheduleCalculationRule: ReviewScheduleCalculationRule): ReviewScheduleWindow =
+    when (reviewScheduleCalculationRule) {
+      ReviewScheduleCalculationRule.PRISONER_READMISSION, ReviewScheduleCalculationRule.PRISONER_TRANSFER -> ReviewScheduleWindow.fromTodayToTenDays()
+      ReviewScheduleCalculationRule.LESS_THAN_6_MONTHS_TO_SERVE -> ReviewScheduleWindow.fromOneToThreeMonths()
+      ReviewScheduleCalculationRule.BETWEEN_6_AND_12_MONTHS_TO_SERVE, ReviewScheduleCalculationRule.PRISONER_ON_REMAND, ReviewScheduleCalculationRule.PRISONER_UN_SENTENCED -> ReviewScheduleWindow.fromTwoToThreeMonths()
+      ReviewScheduleCalculationRule.BETWEEN_12_AND_60_MONTHS_TO_SERVE -> ReviewScheduleWindow.fromFourToSixMonths()
+      ReviewScheduleCalculationRule.MORE_THAN_60_MONTHS_TO_SERVE, ReviewScheduleCalculationRule.INDETERMINATE_SENTENCE -> ReviewScheduleWindow.fromTenToTwelveMonths()
+    }
+
+  private fun reviewScheduleCalculationRuleBasedOnTimeLeftToServe(releaseDate: LocalDate): ReviewScheduleCalculationRule {
+    val today = LocalDate.now()
+    val timeLeftToServe = Period.between(today, releaseDate)
+    val timeLeftToServeInMonths = timeLeftToServe.toTotalMonths()
+    val timeLeftToServeRemainderDays = timeLeftToServe.minusMonths(timeLeftToServeInMonths).days
+
+    return if (timeLeftToServeInMonths < 6 || (timeLeftToServeInMonths == 6L && timeLeftToServeRemainderDays == 0)) {
+      ReviewScheduleCalculationRule.LESS_THAN_6_MONTHS_TO_SERVE
+    } else if (timeLeftToServeInMonths < 12 || (timeLeftToServeInMonths == 12L && timeLeftToServeRemainderDays == 0)) {
+      ReviewScheduleCalculationRule.BETWEEN_6_AND_12_MONTHS_TO_SERVE
+    } else if (timeLeftToServeInMonths < 60 || (timeLeftToServeInMonths == 60L && timeLeftToServeRemainderDays == 0)) {
+      ReviewScheduleCalculationRule.BETWEEN_12_AND_60_MONTHS_TO_SERVE
+    } else {
+      ReviewScheduleCalculationRule.MORE_THAN_60_MONTHS_TO_SERVE
+    }
+  }
+}
+
+enum class SentenceType {
+  RECALL,
+  DEAD,
+  INDETERMINATE_SENTENCE,
+  SENTENCED,
+  CONVICTED_UNSENTENCED,
+  CIVIL_PRISONER,
+  IMMIGRATION_DETAINEE,
+  REMAND,
+  UNKNOWN,
+  OTHER,
+}
+
+data class ReviewScheduleWindow(
+  val dateFrom: LocalDate,
+  val dateTo: LocalDate,
+) {
+  companion object {
+    fun fromTodayToTenDays(): ReviewScheduleWindow = with(LocalDate.now()) { ReviewScheduleWindow(this, this.plusDays(10)) }
```

Comment:
These methods are just reasonably well (?) named convenience methods to construct a `ReviewScheduleWindow` in various different ways.